includes/SVG.php

```diff
@@ -0,0 +1,390 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use SimpleXMLElement;
+use WP_Error;
+use Google\Web_Stories_Dependencies\enshrined\svgSanitize\Sanitizer;
+
+/**
+ * Class SVG
+ *
+ * @package Google\Web_Stories
+ */
+class SVG {
+	/**
+	 * File extenstion.
+	 *
+	 * @var string
+	 */
+	const EXT = 'svg';
+
+	/**
+	 * Mime type.
+	 *
+	 * @var string
+	 */
+	const MIME_TYPE = 'image/svg+xml';
+
+	/**
+	 * Array of SimpleXMLElements.
+	 *
+	 * @var array
+	 */
+	protected $svgs = [];
+
+	/**
+	 * Experiments instance.
+	 *
+	 * @var Experiments Experiments instance.
+	 */
+	private $experiments;
+
+	/**
+	 * SVG constructor.
+	 *
+	 * @param Experiments $experiments Experiments instance.
+	 *
+	 * @return void
+	 */
+	public function __construct( Experiments $experiments ) {
+		$this->experiments = $experiments;
+	}
+
+	/**
+	 * Register filters and actions.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		if ( ! $this->experiments->is_experiment_enabled( 'enableSVG' ) ) {
+			return;
+		}
+
+		add_filter( 'web_stories_allowed_mime_types', [ $this, 'web_stories_allowed_mime_types' ] );
+
+		// Check if svg uploads, already enabled.
+		if ( $this->svg_already_enabled() ) {
+			return;
+		}
+
+		add_filter( 'upload_mimes', [ $this, 'upload_mimes_add_svg' ] ); // phpcs:ignore WordPressVIPMinimum.Hooks.RestrictedHooks.upload_mimes
+		add_filter( 'mime_types', [ $this, 'mime_types_add_svg' ] );
+		add_filter( 'wp_handle_upload_prefilter', [ $this, 'wp_handle_upload' ] );
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'wp_generate_attachment_metadata' ], 10, 3 );
+		add_filter( 'wp_check_filetype_and_ext', [ $this, 'wp_check_filetype_and_ext' ], 10, 5 );
+		add_filter( 'site_option_upload_filetypes', [ $this, 'filter_list_of_allowed_filetypes' ] );
+	}
+
+	/**
+	 * Helper function to check if svg uploads are already enabled.
+	 *
+	 * @return bool
+	 */
+	public function svg_already_enabled() {
+		$allowed_mime_types = get_allowed_mime_types();
+		$mime_types         = array_values( $allowed_mime_types );
+
+		$enabled = in_array( self::MIME_TYPE, $mime_types, true );
+
+		/**
+		 * Filter the check to see if svg support is already enabled.
+		 * This filter allows plugin developers to hotwire this check.
+		 *
+		 * @since 1.2.0
+		 *
+		 * @param bool $enabled Whether the svg support is already enabled.
+		 */
+		return apply_filters( 'web_stories_svg_already_enabled', $enabled );
+	}
+
+	/**
+	 * Enable SVG upload.
+	 *
+	 * @param array $mime_types Mime types keyed by the file extension regex corresponding to those types.
+	 *
+	 * @return mixed
+	 */
+	public function upload_mimes_add_svg( array $mime_types ) {
+		// allow SVG file upload.
+		$mime_types['svg']  = self::MIME_TYPE;
+		$mime_types['svgz'] = self::MIME_TYPE;
+
+		return $mime_types;
+	}
+
+	/**
+	 * Add SVG to allowed mime types.
```

Comment:
```suggestion
	 * Adds SVG to list of mime types and file extensions
```

Since `wp_get_mime_types()` does not deal with allowed mime types, just the list of available/existing ones.